src/Test.EndToEnd.Baselining/Extensions/ArtifactUriEnumeration.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Sarif;
+
+namespace SarifBaseline.Extensions
+{
+    public static class ArtifactUriEnumeration
+    {
+        public static IEnumerable<Uri> AllArtifactUris(this SarifLog log)
+        {
+            if (log != null)
+            {
+                foreach (Run run in log.EnumerateRuns())
+                {
+                    foreach (Uri uri in run.AllArtifactUris())
+                    {
+                        yield return uri;
+                    }
+                }
+            }
+        }
+
+        public static IEnumerable<Uri> AllArtifactUris(this Run run)
+        {
+            if (run?.Artifacts != null)
+            {
+                foreach (Artifact artifact in run.Artifacts)
```

Comment:
>Artifacts [](start = 50, length = 9)

here, of course, you haven't provided run.EnumerateArtifacts. Again, I think our auto-generated code could be helpful. i.e., emit property accessors as above. This would have the benefit of making this value consistent through the OM, rather than for the handful of items we've provided extensions for. Note as well that the code emit machinery could itself be used to autogenerated a consistent set of extension methods, just throwing that out in case it provokes interesting thinking.

Reply:
I think Larry's idea of leveraging the Visitor to accumulate these seems like a good way for the SDK version of this to go. It's very consistent and only requires that the Visitor knows how to walk everything safely. 

To spew a response to another comment here, I also don't think memory use with a Visitor approach will be a problem. One million pointers in a List<Uri> is 8MB. The problem is when each of the things in the list is a new object and they all have 10 pointers in them. Then you're at 92MB.